server/src/test/java/org/opensearch/cluster/metadata/MetadataCreateIndexServiceTests.java

```diff
@@ -1156,228 +1188,204 @@ public void testRemoteStoreNoUserOverrideConflictingReplicationTypeIndexSettings
-            .put(CLUSTER_REMOTE_TRANSLOG_REPOSITORY_SETTING.getKey(), "my-translog-repo-1")
-            .build();
-        FeatureFlagSetter.set(FeatureFlags.REMOTE_STORE);
+    public void testRemoteStoreDisabledByUserIndexSettings() throws Exception {
+        try (FeatureFlagSetter f = FeatureFlagSetter.set(FeatureFlags.REMOTE_STORE)) {
+            Settings settings = Settings.builder()
+                .put(CLUSTER_REPLICATION_TYPE_SETTING.getKey(), ReplicationType.SEGMENT)
+                .put(CLUSTER_REMOTE_STORE_ENABLED_SETTING.getKey(), true)
+                .put(CLUSTER_REMOTE_STORE_REPOSITORY_SETTING.getKey(), "my-segment-repo-1")
+                .put(CLUSTER_REMOTE_TRANSLOG_REPOSITORY_SETTING.getKey(), "my-translog-repo-1")
+                .build();
 
-        request = new CreateIndexClusterStateUpdateRequest("create index", "test", "test");
-        final Settings.Builder requestSettings = Settings.builder();
-        requestSettings.put(SETTING_REMOTE_STORE_ENABLED, false);
-        request.settings(requestSettings.build());
-        Settings indexSettings = aggregateIndexSettings(
-            ClusterState.EMPTY_STATE,
-            request,
-            Settings.EMPTY,
-            null,
-            settings,
-            IndexScopedSettings.DEFAULT_SCOPED_SETTINGS,
-            randomShardLimitService(),
-            Collections.emptySet()
-        );
-        verifyRemoteStoreIndexSettings(indexSettings, "false", null, null, null, null, null);
+            request = new CreateIndexClusterStateUpdateRequest("create index", "test", "test");
+            final Settings.Builder requestSettings = Settings.builder();
+            requestSettings.put(SETTING_REMOTE_STORE_ENABLED, false);
+            request.settings(requestSettings.build());
+            Settings indexSettings = aggregateIndexSettings(
+                ClusterState.EMPTY_STATE,
+                request,
+                Settings.EMPTY,
+                null,
+                settings,
+                IndexScopedSettings.DEFAULT_SCOPED_SETTINGS,
+                randomShardLimitService(),
+                Collections.emptySet()
+            );
+            verifyRemoteStoreIndexSettings(indexSettings, "false", null, null, null, null, null);
+        }
     }
 
-    public void testRemoteStoreTranslogDisabledByUserIndexSettings() {
-        Settings settings = Settings.builder()
-            .put(CLUSTER_REPLICATION_TYPE_SETTING.getKey(), ReplicationType.SEGMENT)
-            .put(CLUSTER_REMOTE_STORE_ENABLED_SETTING.getKey(), true)
-            .put(CLUSTER_REMOTE_STORE_REPOSITORY_SETTING.getKey(), "my-segment-repo-1")
-            .put(CLUSTER_REMOTE_TRANSLOG_REPOSITORY_SETTING.getKey(), "my-translog-repo-1")
-            .build();
-        FeatureFlagSetter.set(FeatureFlags.REMOTE_STORE);
+    public void testRemoteStoreTranslogDisabledByUserIndexSettings() throws Exception {
```

Comment:
Might be good to use 
```
@Before
    public void setUp() {
    //blah
    }

```
rather than redundantly placing the logic at every test?

Reply:
The feature flag is only set for very few tests. We don't need the feature flag to be set for all the tests.

Reply:
Did something similar as suggested, @Bukhtawar.